src/geometry/collider_set.rs

```diff
@@ -66,6 +66,7 @@ impl ColliderSet {
         self.modified_colliders.clear();
         let modified_colliders = &mut self.modified_colliders;
         self.colliders.iter_mut().map(move |(h, b)| {
+            b.changes |= ColliderChanges::MODIFIED;
```

Comment:
I'm not sure this is correct ; the fix may be more around `https://github.com/Vrixyz/rapier/blob/607f0988596d346b6afa70e5cfbc8f0f3519e02c/src/geometry/collider.rs#L271` ; to set `ColliderChanges::MODIFIED` flag too (see https://github.com/dimforge/rapier/pull/803/files#r1980964215 comment for a longer term fix maybe ?)

And then add then refill `self.modified_colliders` only when applicable (check `changes` flag)